views/default/elgg.js.php

```diff
@@ -51,11 +51,13 @@
 	echo "\n";
 }
 
-/**
- * Set some values that are cacheable
- */
+// Set some values that are cacheable
+$data = (object)elgg_trigger_plugin_hook('elgg.data', 'site', null, []);
 ?>
 //<script>
+// page data overrides site data
+elgg.data = $.extend(<?= json_encode($data) ?>, elgg._data);
+delete elgg._data;
 
 elgg.version = '<?php echo elgg_get_version(); ?>';
```

Comment:
Maybe add `_elgg_get_init_client_site_data()` and rename the other to `_elgg_get_init_client_page_data()`?